packages/core/src/external/opensearch/index.ts

```diff
@@ -1,36 +1,25 @@
-/**
- * Created outside of src/external/aws because this is an open source technology
- * that could be hosted outside of AWS.
- */
-import { DocumentReference } from "@medplum/fhirtypes";
-
 export const contentFieldName = "content";
 
-// TODO ENG-41 this is "file" specific and should be moved to the file-ingestor.ts file
-export type IndexFields = {
-  cxId: string;
-  patientId: string;
-  s3FileName: string;
-  [contentFieldName]: string;
-};
-
-export function isDocStatusReady(doc: DocumentReference): boolean {
-  return !isDocStatusPreliminary(doc) && !isDocStatusEnteredInError(doc);
-}
-export function isDocStatusFinal(doc: DocumentReference): boolean {
-  return doc.docStatus === "final";
-}
-export function isDocStatusAmended(doc: DocumentReference): boolean {
-  return doc.docStatus === "amended";
-}
-export function isDocStatusPreliminary(doc: DocumentReference): boolean {
-  return doc.docStatus === "preliminary";
-}
-export function isDocStatusEnteredInError(doc: DocumentReference): boolean {
-  return doc.docStatus === "entered-in-error";
-}
-
-export type OpenSearchIngestorConfig = {
+export type OpenSearchConfig = {
   region: string;
   indexName: string;
 };
+export type OpenSearchConfigDirectAccess = OpenSearchConfig & {
+  endpoint: string;
+  username: string;
+  password: string;
+};
+
+// https://github.com/opensearch-project/opensearch-js/issues/269
+export type OpenSearchResponseHit<T> = {
+  _index: string;
+  _id: string;
+  _score: number;
+  _source: T;
+};
+// https://github.com/opensearch-project/opensearch-js/issues/269
+export type OpenSearchResponse<T> = {
+  hits: {
+    hits?: OpenSearchResponseHit<T>[];
+  };
+};
```

Comment:
Moved from packages/core/src/external/opensearch/semantic/semantic-searcher.ts w/ addition of Generics